core/src/main/java/com/yahoo/oak/BlocksPool.java

```diff
@@ -22,33 +22,50 @@
 
     // TODO change the following constants to be configurable
 
-    // Size of a single memory block - currently 256MB
-    static final int BLOCK_SIZE = 256 * 1024 * 1024;
+    static final long MB = 1L << 20;
+    static final long GB = 1L << 30;
 
-    // Number of memory blocks to be pre-allocated
-    private static final int PRE_ALLOC_BLOCKS = 0;
+    // The memory size to pre-allocate on initialization
+    private static final long PRE_ALLOC_SIZE_BYTES = 0;
 
-    // Number of memory blocks to be allocated at once when not enough blocks available
-    private static final int NEW_ALLOC_BLOCKS = 1;
+    // The memory size to be allocate at once when not enough memory is available
+    private static final long NEW_ALLOC_SIZE_BYTES = 8L * MB;
```

Comment:
Why 8MB? Isn't it the final version? This change was in order to leave default as it was or 128MB and make the change for Druid...

Reply:
I am missing something here... How blocks of 256MB is going to fit into 8MB? Is it because of the mathematical maximum always only one block is about to be allocated? I don't think it should be done this way...

Reply:
It is actually the minimal size of a new allocation. It can be larger if the block size is larger.
I can change the name of this constant to `NEW_ALLOC_MIN_SIZE_BYTES` and change the comment as well.

> I don't think it should be done this way...

How do you think it should be?

Reply:
You can proceed with your suggestion:

>  I can change the name of this constant to NEW_ALLOC_MIN_SIZE_BYTES and change the comment as well.

Or to make NEW_ALLOC_SIZE_BYTES be equal to DEFAULT_BLOCK_SIZE_BYTES and change if blockSizeBytes is different.

Reply:
Your suggestion nullifies this parameter altogether. It means that the number of allocated blocks is always 1.
Actually, they were always meaningless after my previous set of PRs. We left these parameters in the code only to allow us to fiddle with them for testing, but in production, they are actually useless.
So for now I'll go with `NEW_ALLOC_MIN_SIZE_BYTES`, but we can further discuss them as part of your new memory-manager design review.